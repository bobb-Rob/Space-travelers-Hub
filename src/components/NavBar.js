const NavBar = () => <header><nav>Navigation bar</nav></header>;

export default NavBar;
