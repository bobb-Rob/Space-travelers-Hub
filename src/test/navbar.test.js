import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavBar from '../components/NavBar';
// import NavLinkEl from '../components/NavLinkEl';

test('Test for text contents in links', async () => {
  render(<NavBar />, { wrapper: Router });

  const textElement = screen.getByText('Rockets');
  expect(textElement).toBeInTheDocument();
});

it('Builds the snapchot of the navbar component safely', () => {
  const tree = renderer
    .create(
      <Router>
        <NavBar />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
