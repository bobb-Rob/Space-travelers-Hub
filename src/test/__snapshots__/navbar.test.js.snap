// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Builds the snapchot of the navbar component safely 1`] = `
<header>
  <nav
    className="nav-container"
  >
    <div
      className="logo-wrapper"
    >
      <img
        alt=""
        className="image-logo"
        src="../images/saturn.png"
      />
      <h2>
        Space Traveler's Hub
      </h2>
    </div>
    <div>
      <a
        aria-current="page"
        className="nav-link active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "underline",
          }
        }
      >
        Rockets
      </a>
      <a
        className="nav-link"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
      |
      <a
        className="nav-link"
        href="/profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </div>
  </nav>
</header>
`;
